### DISCUSSION TITLE
RESTORE after nonlocal means denoising

### DISCUSSION THREAD

**Discussion by salma1601** [dipy/dipy#1175] [Reconstruction]
Hi Dipy experts,

Is it possible use the 'RESTORE' fit_method for DTI reconstruction after having denoised my DWI with non-local means ? Or is advised not to denoise the data ?

**Comment by samuelstjean** [comment]
That's a good question, I never tried residual noise estimation after a first pass. Since you need to supply some estimate of the variance to it to find the outliers, you can't use the original value you supplied to nlmeans nor use any of the two functions I know about to estimate noise.

I'm just tagging along for another answer, I am also curious since I never thought about this case, I only used the recursive calibration method (rough equivalent for fodf) which autotunes itself internally).

**Comment by salma1601** [comment]
Well I am doing restore -> denoise -> EstimateResponseSH -> CSD -> StreamlineTractography
and it looks like my streamlines are better with the denoise step than without, but I was concerned about using the Tensor fitting outputs in the pipeline together with a denoised DWI, since these outputs are not generated with the denoised DWI

**Comment by samuelstjean** [comment]
Not sure I follow, you mean you run restore -> predict back the signal -> nlmeans then the rest?

**Comment by salma1601** [comment]
I mean that in the inputs of EstimateResponseSH I am giving as inputs  
- denoised DWI
- evals from RESTORE fitted on the **original non denoised ** image.
and I am not sure if this makes sense.
Sorry I am new to dipy and diffusion and I am not aware if my question makes sense. I'am using dipy interfaced by **nipype** and trying to make a whole cached pipeline but I am afraid of passing inputs that are not compatible together
